### ISSUE TITLE
Test suite failure with QuickCheck 2.9

### ISSUE THREAD

**Issue by snoyberg** [fpco/inline-c#51]
```
Failures:

  test/Language/C/Types/ParseSpec.hs:26: 
  1) Language.C.Types.Parse parses everything which is pretty-printable (C)
       Gave up after 0 tests

  test/Language/C/Types/ParseSpec.hs:33: 
  2) Language.C.Types.Parse parses everything which is pretty-printable (Haskell)
       Gave up after 0 tests

Randomized with seed 1266486728
```

**Comment by bitonic** [comment]
Thanks, I'll try to reproduce.

**Comment by bitonic** [comment]
With the resolver in the repo, `lts-6.7`, things work with that seed (using `stack test inline-c --test-arguments '--seed=1266486728'`). Nightly works as well (`stack test --resolver=nightly inline-c --test-arguments '--seed=1266486728'`).

Waiting for the newer nightly to try again, or other suggestions by @snoyberg on how to reproduce.

**Comment by bitonic** [comment]
Reproduced by adding `QuickCheck-2.9` in extra-deps.

**Comment by bitonic** [comment]
This is due to https://github.com/nick8325/quickcheck/issues/113 , which is fixed in the latest QuickCheck and should be fixed in QuickCheck-2.9.2.

I've verified that the bug goes away when compiling with the latest `master`.
